src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ConsiderPassingBaseTypesAsParameters.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
+{
+    using static MicrosoftCodeQualityAnalyzersResources;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class ConsiderPassingBaseTypesAsParameters : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1011";
+
+        internal static readonly DiagnosticDescriptor Rule =
+            DiagnosticDescriptorHelper.Create(
+                RuleId,
+                CreateLocalizableResourceString(ConsiderPassingBaseTypesAsParametersTitle),
+                CreateLocalizableResourceString(ConsiderPassingBaseTypesAsParametersMessage),
+                DiagnosticCategory.Design,
+                RuleLevel.IdeSuggestion,
+                description: CreateLocalizableResourceString(ConsiderPassingBaseTypesAsParametersDescription),
```

Comment:
I think this should fix it :)

```suggestion
                CreateLocalizableResourceString(nameof(ConsiderPassingBaseTypesAsParametersTitle)),
                CreateLocalizableResourceString(nameof(ConsiderPassingBaseTypesAsParametersMessage)),
                DiagnosticCategory.Design,
                RuleLevel.IdeSuggestion,
                description: CreateLocalizableResourceString(nameof(ConsiderPassingBaseTypesAsParametersDescription)),
```

Reply:
I am so stupiiiidd! Thank you so much :D

Reply:
This took time first time I hit it. I wish a "constant parameter" feature in C#, which would have avoided this and I think helps occasionally. Not sure if there is already a proposal.

Reply:
I saw something about it a couple of days ago